erpnext/accounts/report/profit_and_loss_statement/profit_and_loss_statement.py

```diff
@@ -23,20 +25,91 @@ def execute(filters=None):
 		accumulated_values=filters.accumulated_values,
 		ignore_closing_entries=True, ignore_accumulated_values_for_fy=ignore_accumulated_values_for_fy)
 
-	net_profit_loss = get_net_profit_loss(income, expense, period_list, filters.company, filters.presentation_currency)
-
+	#add the section for income
 	data = []
 	data.extend(income or [])
-	data.extend(expense or [])
+
+	#get direct and indirect expense account names
+	gross_profit_loss = None
+	direct_expense_account = frappe.get_value("Company", get_default_company(), "default_direct_expenses")
+	indirect_expense_account = frappe.get_value("Company", get_default_company(), "default_indirect_expenses")
+	direct_expenses = None
+	for expense_item in expense:
+		data.append(expense_item)
+
+		#add gross profit line item on the PL statement
+		if expense_item.get("account_name") == "Total  "+direct_expense_account.split("-")[1].strip():
```

Comment:
- Is there no better way to do this ?

Reply:
There has to be a better way of doing this!
Add a key in expense dict for comparing accounts rather than this